### PR TITLE
test(strategy): serialize and deserialize User

### PR DESCRIPTION
**Pull request by udaydenduluri33**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/auth/models/strategy.class.ts [hunk 1]
```diff
@@ -246,4 +238,12 @@ export abstract class Strategy extends events.EventEmitter {
     public getEvents = (): string[] => {
         return Object.values<string>(AUTH.EVENT)
     }
+
+    public emitIfListenersExist(eventObject: unknown, done: (err: any, id?: unknown) => void, eventName: string) {
```
Comment (paddybasi) [inline src/auth/models/strategy.class.ts:242]: could you put the eventName first, then perhaps use the spread operator to allow for any number of arguments? This way the function could be more generic and applied everywhere

Comment (udaydenduluri33) [reply src/auth/models/strategy.class.ts:243]: ok

Comment (udaydenduluri33) [reply src/auth/models/strategy.class.ts]: @paddybasi  done as requested

File: src/auth/models/strategy.class.ts [hunk 2]
```diff
@@ -246,4 +239,12 @@ export abstract class Strategy extends events.EventEmitter {
     public getEvents = (): string[] => {
         return Object.values<string>(AUTH.EVENT)
     }
+
+    public emitIfListenersExist(eventName: string, eventObject: unknown, done: (err: any, id?: unknown) => void) {
```
Comment (paddybasi) [inline src/auth/models/strategy.class.ts:243]: jsdoc style comments please

File: src/auth/models/strategy.class.ts [hunk 3]
```diff
@@ -31,9 +31,10 @@ export abstract class Strategy extends events.EventEmitter {
         tokenEndpointAuthMethod: '',
     }
 
-    protected constructor(strategyName: string) {
+    protected constructor(strategyName: string, router: Router) {
```
[the remark this answers is not in the document]
Comment (udaydenduluri33) [reply src/auth/models/strategy.class.ts:34]: we are doing it in the derived class. This is an abstract class

File: src/auth/models/strategy.class.ts (hunk 2 above)
Comment (udaydenduluri33) [reply src/auth/models/strategy.class.ts:243]: Done @paddybasi

**Pull request merged**